tests_fp/solidity/test_close_short.py

```diff
@@ -76,32 +77,33 @@ def verify_close_short(
         # verify that the bond reserves were updated according to flat+curve
         # the adjustment should be equal to timeRemaining * bondAmount
         self.assertLess(  # user gets less than what they put in
-            agent_base_proceeds,
-            agent_base_paid,
+            int(agent_base_proceeds),
+            int(agent_base_paid),
             msg="agent gets more than what they put in: agent_base_proceeds > agent_base_paid",
         )
-        time_remaining = 0
+        time_remaining = FixedPoint("0.0")
         if maturity_time > self.hyperdrive.block_time.time:
             time_remaining = maturity_time - self.hyperdrive.block_time.time
-        stretch_time = StretchedTime(
-            days=time_remaining * 365, time_stretch=self.hyperdrive.time_stretch_constant, normalizing_constant=365
-        )
-        model = yieldspace_pm.YieldspacePricingModel()
-        curve_shares = float(
-            model.calc_shares_in_given_bonds_out(
-                Decimal(market_state_before.share_reserves),
-                Decimal(market_state_before.bond_reserves),
-                Decimal(market_state_before.lp_total_supply),
-                Decimal(bond_amount * time_remaining),
-                Decimal(1 - stretch_time.stretched_time),
-                Decimal(market_state_before.share_price),
-                Decimal(market_state_before.init_share_price),
-            )
+        stretch_time = StretchedTimeFP(
+            days=time_remaining * FixedPoint("365.0"),
+            time_stretch=self.hyperdrive.time_stretch_constant,
+            normalizing_constant=FixedPoint("365.0"),
         )
-        flat_shares = bond_amount * (1 - time_remaining) / market_state_before.share_price
-        np.testing.assert_allclose(
-            self.hyperdrive.market_state.share_reserves + flat_shares + curve_shares,
+        model = yieldspace_pm.YieldspacePricingModelFP()
+        curve_shares = model.calc_shares_in_given_bonds_out(
             market_state_before.share_reserves,
+            market_state_before.bond_reserves,
+            market_state_before.lp_total_supply,
+            bond_amount * time_remaining,
+            FixedPoint("1.0") - stretch_time.stretched_time,
+            market_state_before.share_price,
+            market_state_before.init_share_price,
+        )
+
+        flat_shares = bond_amount * (FixedPoint("1.0") - time_remaining) / market_state_before.share_price
+        np.testing.assert_allclose(
```

Comment:
I think we should remove all `np.testing` calls and replacing with `self.assertAlmostEqual` to stay consistent with the `unittest` stdlib. I've been doing this in my FixedPoint refactored tests.

Reply:
i think delta is better with ints because we know exactly how much it is off.  places can be dynamic with floats and doesn't work with int/fp